lib/actions/addLinkToProjectAction.js

```diff
@@ -11,15 +11,20 @@ class AddLinkToProjectAction {
   parseParams(userInputString = '') {
     assert(typeof userInputString === 'string');
     const parsedUserInput = userInputString.split(' ');
-
+    let linkParams;
     if (parsedUserInput.length < 2) {
       throw new TypeError('invalid input');
     }
 
-    const linkParams = {
-      label: parsedUserInput[1],
-      link: parsedUserInput[2],
-    };
+    if (parsedUserInput.length >= 4) {
+      linkParams = {
+        label: parsedUserInput.slice(1, 3).join(' '),
```

Comment:
This method is on the right track, but will only work if 3 words are used for the label. If only 1 or 2 words are used for the label, then the `slice()` will end up including the URL. If 4+ words are used for the label, then only the first 3 will be included in the label.

This function would a great candidate for test-driven coding. Write your test cases first with multiple scenarios you would expect this to work with, then change your code to make the tests pass.

Reply:
I'm also concerned that his approach to accepting multiple words will cause future problems. What happens when you add a new parameter that can also be multiple words? How will you be able to distinguish between where one parameter ends and the other one starts?

Maybe you could support this method at first, then look at another method (ie. multi-word params in quotes) in the future.